### ISSUE TITLE
Move VS4Mac off of SolutionCrawler

### ISSUE THREAD

**Issue by CyrusNajmabadi** [dotnet/roslyn#64839]
There appears to be one incremental-analyzer vs4mac is still using.  From offline discussions, this type does not seem to be necessary.  It needs to be removed so we can remove solutioncrawler.

**Comment by davidwengier** [comment]
@sandyarmstrong @Therzok Did we get confirmation that the project telemetry isn't needed, and the analyzer can be removed?

**Comment by sandyarmstrong** [comment]
> @sandyarmstrong @Therzok Did we get confirmation that the project telemetry isn't needed, and the analyzer can be removed?

Yes, that was my understanding from https://teams.microsoft.com/l/message/19:634f57b2228a47f4b75dec5b5276ca86@thread.tacv2/1665104245532?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3598d109-00b5-420c-9e48-0bfd087b65c0&parentMessageId=1665104245532&teamName=Visual%20Studio%20for%20Mac&channelName=Technical%20Insights%20%F0%9F%A7%90&createdTime=1665104245532&allowXTenantAccess=false .

**Comment by Therzok** [comment]
Yep, that is correct.

**Comment by davidwengier** [comment]
There are also preferences for background scope analysis, using SolutionCrawlerOptionsStorage. Does that just go away? Or is there a replacement?

**Comment by CyrusNajmabadi** [comment]
I would think that would just go away

**Comment by davidwengier** [comment]
TypeScript uses Solution Crawler on VS Mac here:
https://devdiv.visualstudio.com/DevDiv/_git/TypeScript-VS?path=/VS/LanguageService/TypeScriptVSMac/Addin/TextViewCreationListener.cs&version=GBmain&line=28&lineEnd=29&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

**Comment by CyrusNajmabadi** [comment]
@davidwengier terrific.  Do you know what that crawler is for?  @Therzok ?

**Comment by Therzok** [comment]
I don't know anything about the typescript extension.

**Comment by MariaSolOs** [comment]
So I ***think*** that we use the solution crawler in that text view listener to properly synchronize the TypeScript project and Roslyn workspace when a .ts file (including script blocks) gets created.

**Comment by MariaSolOs** [comment]
Those kind of synchronization methods are also used in our integration tests to make sure that the language service is ready before continuing with the actual test verification.

Also pinging @zkat since they do most of our VSMac work.

**Comment by CyrusNajmabadi** [comment]
> So I think that we use the solution crawler in that text view listener to properly synchronize the TypeScript project and Roslyn workspace when a .ts file (including script blocks) gets created.

We def need more information on this :)

**Comment by davidwengier** [comment]
There is a branch available on VS Mac that has no solution crawler usage in it: https://github.com/xamarin/vsmac/pull/9272

For anyone with a mac, builds are available from that PR for testing. For anyone without a mac (ie, @CyrusNajmabadi 😛) is there a list of things that we should validate, to make sure no user experiences has regressed? Or was there already no active use of Solution Crawler for important functionality in Roslyn?

**Comment by seoworl** [comment]
i realy need more info on that

**Comment by davidwengier** [comment]
Note: With https://github.com/xamarin/vsmac/pull/10182 we've had to add a new usage of solution crawler, for the task list analyzers.

**Comment by CyrusNajmabadi** [comment]
Closing out.  Solution crawler is gone.  As is vs4mac.

**Comment by MariaSolOs** [comment]
> Closing out. Solution crawler is gone. As is vs4mac.

This is my favorite issue resolution 😸
